libsel4/tools/syscall_stub_gen.py

```diff
@@ -1103,8 +104,10 @@ def main():
         print("Invalid word size.")
         sys.exit(2)
 
-    # Generate the stubs.
-    generate_stub_file(args.arch, wordsize, args.files, args.output, args.buffer, args.mcs)
+    pick_generator = {'c': CGenerator, 'rust': RustGenerator}
```

Comment:
Maybe we should delegate this to a a helper class to remove it from here. It's a long term bet that there might be more, The heller would know the available generators and and could also provide the list `choices` for the `--lang` above.

Reply:
Yes, now we have `GeneratorProvider` class.
It stores information about available generators, provides a list of their names (for `choices`), and allows retrieving a generator using its name. Additionally there's a custom `Exception`, and in case wrong generator name was given, it prints available generators.